classes/services/queryBuilders/PKPStatsSushiQueryBuilder.inc.php

```diff
@@ -0,0 +1,225 @@
+                    } else {
+                        $selectColumns[$i] = DB::raw('YEAR(STR_TO_DATE(p.date_published, "%Y-%m-%d")) as YOP');
+                    }
+                    break;
+                }
+            }
+        }
+
+        // Build the select and group by clauses.
+        if (!empty($selectColumns)) {
+            $q->select($selectColumns);
+            if (!empty($groupBy)) {
+                $q->groupBy($groupBy);
+            }
+        }
+        $counterMetricsColumns = StatisticsHelper::getCounterMetricsColumns();
+        foreach ($counterMetricsColumns as $counterMetricsColumn) {
+            $q->addSelect(DB::raw("SUM({$counterMetricsColumn}) AS {$counterMetricsColumn}"));
+        }
+        return $q;
+    }
+
+    /**
+     * @copydoc PKPStatsQueryBuilder::_getObject()
+     */
+    protected function _getObject(): \Illuminate\Database\Query\Builder
+    {
+        if ($this->institutionId === 0) {
+            // consider only monthly DB table
+            $q = DB::table('metrics_counter_submission_monthly as m');
+        } else {
+            // consider only monthly DB table
+            $q = DB::table('metrics_counter_submission_institution_monthly as m');
+        }
+
+        if (!empty($this->yearsOfPublication)) {
+            $q->leftJoin('publications as p', function ($q) {
+                $q->on('p.submission_id', '=', 'm.submission_id')
+                    ->where('p.version', '=', 1);
+            });
+            foreach ($this->yearsOfPublication as $yop) {
+                if (preg_match('/\d{4}/', $yop)) {
+                    if (substr(Config::getVar('database', 'driver'), 0, strlen('postgres')) === 'postgres') {
+                        $q->where(DB::raw('EXTRACT(YEAR FROM p.date_published)'), '=', $yop);
+                    } else {
+                        $q->where(DB::raw('YEAR(STR_TO_DATE(p.date_published, "%Y-%m-%d"))'), '=', $yop);
+                    }
+                } elseif (preg_match('/\d{4}-\d{4}/', $yop)) {
+                    $years = explode('-', $yop);
+                    if (substr(Config::getVar('database', 'driver'), 0, strlen('postgres')) === 'postgres') {
+                        $q->whereBetween(DB::raw('EXTRACT(YEAR FROM p.date_published)'), $years);
+                    } else {
+                        $q->whereBetween(DB::raw('YEAR(STR_TO_DATE(p.date_published, "%Y-%m-%d"))'), $years);
+                    }
+                }
+            }
+        }
+
+        if (!empty($this->contextIds)) {
+            $q->whereIn('m.' . StatisticsHelper::STATISTICS_DIMENSION_CONTEXT_ID, $this->contextIds);
+        }
+
+        if (!empty($this->submissionIds)) {
+            $q->whereIn('m.' . StatisticsHelper::STATISTICS_DIMENSION_SUBMISSION_ID, $this->submissionIds);
+        }
+
+        $q->whereBetween('m.' . StatisticsHelper::STATISTICS_DIMENSION_MONTH, [date_format(date_create($this->dateStart), 'Ym'), date_format(date_create($this->dateEnd), 'Ym')]);
```

Comment:
This is checked in PKPStatsSushiHandler. If the beginDate is not the 1. of the month and/or if the endDate is not the last day of a month, a warning will be created and dates corrected, so that the beginDate is the 1. of the month and endDate the last date of the month.